engine/lib/phx/src/engine/main_loop.rs

```diff
@@ -0,0 +1,268 @@
+use mlua::Function;
+use std::path::PathBuf;
+use winit::application::ApplicationHandler;
+use winit::event::{self, *};
+use winit::event_loop::ActiveEventLoop;
+use winit::keyboard::PhysicalKey;
+use winit::window::WindowId;
+
+use super::Engine;
+use crate::window::*;
+use glam::*;
+use tracing::*;
+
+pub struct MainLoop {
+    pub engine: Option<Engine>,
```

Comment:
Doing unwrap each frame feels not very nice. Would it be possible to have some 'default' engine instead?
Otherwise I'd prefer to use `expect`/`unwrap_or_else` instead of `unwrap`.

Reply:
A default engine would mean we would have to push all the Option<...>'s into the Engine itself. I think the only part of it that needs to be deferred is the Window, so maybe we can get away with making that an Option<...>.

I know that unwrap is not very nice, however Winit guarantees that the Init event will always be the first one, so I think I would prefer to leave it as `expect`, so at least the crash is more obvious.

Reply:
OK, I replaced the unwraps with:

```
        let Some(engine) = self.engine.as_mut() else { return; };
```

So at least this time, if winit decides to call a method of the trait, we just exit from the method rather than crash.

Reply:
> A default engine would mean we would have to push all the Option<...>'s into the Engine itself. I think the only part of it that needs to be deferred is the Window, so maybe we can get away with making that an Option<...>.

Yes, this probably would be better but check it yourself - if it's too complex, then I'm ok with current implementation too.